slides/lesson2/solution/main.tf

```diff
@@ -91,12 +113,12 @@ resource "aws_security_group" "web" {
 #--------------------------------------------------------------
 resource "aws_launch_configuration" "lc" {
   name_prefix                 = "${var.metadata["appname"]}-${var.env}-lc-${var.metadata["appversion"]}-"
-  image_id                    = "${data.aws_ami.amazon_linux.id}"
-  instance_type               = "${var.instance_type}"
-  security_groups             = ["${aws_security_group.web.id}"]
-  user_data                   = "${data.template_file.deploy_sh.rendered}"
-  key_name                    = "${var.key_name}"
-  associate_public_ip_address = "${var.associate_public_ip_address}"
+  image_id                    = data.aws_ami.amazon_linux.id
+  instance_type               = var.instance_type
+  security_groups             = [aws_security_group.web.id]
+  user_data                   = data.template_file.deploy_sh.rendered
+  associate_public_ip_address = var.associate_public_ip_address
+  spot_price                  = "0.02"
```

Comment:
What are your thoughts on removing the spot price? My assumption is that these instances will only be alive for a few minutes. Otherwise, we may run into issues where participants could get their instances terminated if we have bad luck and the spot price increases above this value

Reply:
we need to keep this as low as possible in price. This can be managed very easily just by commenting out this line.